client/X11/xf_rail.c

```diff
@@ -1101,18 +1101,30 @@ int xf_rail_init(xfContext* xfc, RailClientContext* rail)
 	if (!xfc->railWindows)
 		return 0;
 
-	xfc->railWindows->keyCompare = rail_window_key_equals;
-	xfc->railWindows->hash = rail_window_key_hash;
-	xfc->railWindows->valueFree = rail_window_free;
+	if (!HashTable_SetHashFunction(xfc->railWindows, rail_window_key_hash))
+		goto fail;
+	{
+		wObject* obj = HashTable_KeyObject(xfc->railWindows);
+		if (!obj)
```

Comment:
That should be an assert, if `HashTable_KeyObject` returns `NULL` it means we're in big trouble.

Reply:
since this is external of `winpr` I´d keep the error check. will modify the internals for `HashTable_SetupForStringData` though